packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -0,0 +1,200 @@
+			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+			event.stopPropagation();
+		}
+	};
+
+	const selectMedia = ( media ) => {
+		onSelect( media );
+		setMediaURLValue( media.url );
+		speak( __( 'The media file has been replaced' ) );
+	};
+
+	const uploadFiles = ( event ) => {
+		const files = event.target.files;
+		const setMedia = ( [ media ] ) => {
+			setShowMediaReplaceOptions( false );
+			selectMedia( media );
+		};
+		mediaUpload( {
+			allowedTypes,
+			filesList: files,
+			onFileChange: setMedia,
+			onError,
+		} );
+	};
+
+	const onClose = () => {
+		editMediaButtonRef.current.focus();
+	};
+
+	const onClickOutside = () => ( setShowMediaReplaceOptions( false ) );
+
+	const openOnArrowDown = ( event ) => {
+		if ( event.keyCode === DOWN ) {
+			event.preventDefault();
+			event.stopPropagation();
+			event.target.click();
+		}
+	};
+
+	let urlInputUIContent;
+	if ( showEditURLInput ) {
+		urlInputUIContent = (
+			<LinkEditor
+				onKeyDown={ stopPropagationRelevantKeys }
+				onKeyPress={ stopPropagation }
+				value={ mediaURLValue }
+				isFullWidthInput={ true }
+				hasInputBorder={ true }
+				onChangeInputValue={ ( url ) => ( setMediaURLValue( url ) ) }
+				onSubmit={ ( event ) => {
+					event.preventDefault();
+					onSelectURL( mediaURLValue );
+					setShowEditURLInput( ! showEditURLInput );
+					editMediaButtonRef.current.focus();
+				} }
+			/>
+		);
+	} else {
+		urlInputUIContent = (
+			<LinkViewer
+				isFullWidth={ true }
+				className="editor-format-toolbar__link-container-content block-editor-format-toolbar__link-container-content"
+				url={ mediaURLValue }
+				onEditLinkClick={ () => ( setShowEditURLInput( ! showEditURLInput ) ) }
+			/>
+		);
+	}
+
+	return (
+		<MediaUpload
+			onSelect={ ( media ) => selectMedia( media ) }
+			onClose={ () => setShowMediaReplaceOptions( true ) }
+			allowedTypes={ allowedTypes }
+			render={ ( { open } ) => (
+				<Toolbar className={ 'media-replace-flow components-dropdown-menu' }>
+					<Button
+						ref={ editMediaButtonRef }
+						className={ 'components-icon-button components-dropdown-menu__toggle' }
+						onClick={ () => {
+							setShowMediaReplaceOptions( ! showMediaReplaceOptions );
+						} }
+						onKeyDown={ openOnArrowDown }
+					>
+						<span className="components-dropdown-menu__label" > { name } </span>
+						<span className="components-dropdown-menu__indicator" />
+					</Button>
+					{ showMediaReplaceOptions &&
+						<Popover
+							onClickOutside={ onClickOutside }
+							onClose={ onClose }
+							className={ 'media-replace-flow__options' }
+						>
+							<>
+								<NavigableMenu>
+									<MenuItem
+										icon="admin-media"
+										onClick={ open }
+									>
+										{ __( 'Open Media Library' ) }
+									</MenuItem>
+									<MediaUploadCheck>
+										<FormFileUpload
+											onChange={ uploadFiles }
+											accept={ allowedTypes }
```

Comment:
Currently, we have a bug, when you press the upload button the window to select the file does not list only images but lists everything.

allowedTypes is the format used by the wp media library to filter file types. But the accept property used by browsers is different e.g: "image/*".
MediaPlaceholder allows the user to pass allowed types and accept, I guess this component should also accept both properties and on the image, we can pass "image/*" as the accept value.

Reply:
Thank you @jorgefilipecosta I've cleaned up the ImageEdit component for all the editing state functions and props and have also fixed the accepts bug in the upload dialog :)